src/transformers/data/data_collator.py

```diff
@@ -996,15 +996,15 @@ def tf_mask_tokens(self, inputs: Any, mask_labels: Any) -> Tuple[Any, Any]:
             raise ValueError(
                 "This tokenizer does not have a mask token which is necessary for masked language modeling. Remove the --mlm flag if you want to use this tokenizer."
             )
-        labels = inputs.clone()
+        labels = tf.identity(inputs)
         # We sample a few tokens in each sequence for masked-LM training (with probability args.mlm_probability defaults to 0.15 in Bert/RoBERTa)
 
         masked_indices = tf.cast(mask_labels, tf.bool)
 
         special_tokens_mask = [
-            self.tokenizer.get_special_tokens_mask(val, already_has_special_tokens=True) for val in labels.tolist()
+            self.tokenizer.get_special_tokens_mask(val, already_has_special_tokens=True) for val in labels
         ]
-        masked_indices = masked_indices & ~tf.convert_to_tensor(special_tokens_mask, dtype=tf.bool)
+        masked_indices = masked_indices & ~tf.cast(special_tokens_mask, dtype=tf.bool)
```

Comment:
Does this change assume the tokenizer is set to `return_tensors="tf"`?

Reply:
Nope, TensorFlow coerces the python/numpy value into a Tensor here:

```
>>> tf.cast([0, 1, 0, 1], dtype=tf.bool)
<tf.Tensor: shape=(4,), dtype=bool, numpy=array([False,  True, False,  True])>
>>> tf.cast(np.array([0, 1, 0, 1]), dtype=tf.bool)
<tf.Tensor: shape=(4,), dtype=bool, numpy=array([False,  True, False,  True])>
```

This behavior is also tested with whatever the default `return_tensors` (looks like they are just lists)
https://github.com/huggingface/transformers/pull/13891/files/94a3cf08f1e328202739233b1d58980edcddc2af#diff-3d06edcb84efb377e8c1b5545f396cdf69a1de21ff82cbf2aab3902e4b42dec5R505-R506

Reply:
Ah, that works! The default return_tensors does indeed just return Python lists.